lib/elixir/lib/calendar/iso.ex

```diff
@@ -742,14 +777,23 @@ defmodule Calendar.ISO do
           Calendar.microsecond(),
           :basic | :extended
         ) :: String.t()
-  def time_to_string(hour, minute, second, microsecond, format \\ :extended)
+  def time_to_string(
+        hour,
+        minute,
+        second,
+        {ms_value, ms_precision} = microsecond,
+        format \\ :extended
+      )
+      when is_hour(hour) and is_minute(minute) and is_second(second) and
+             is_microsecond(ms_value, ms_precision) and format in [:basic, :extended] do
+    time_to_string_guarded(hour, minute, second, microsecond, format)
+  end
 
-  def time_to_string(hour, minute, second, {_, 0}, format) when format in [:basic, :extended] do
+  def time_to_string_guarded(hour, minute, second, {_, 0}, format) do
```

Comment:
Make private?

Reply:
thank you @clediano. All 4 clauses have been made private